manifests/ssltools/nssdb.pp

```diff
@@ -1,50 +1,30 @@
 # Sets up nssdb
 class certs::ssltools::nssdb (
-  $nss_db_dir = $certs::nss_db_dir,
+  $pki_dir = $certs::pki_dir,
```

Comment:
I'd consider changing this:
```suggestion
  $nss_db_dir = "${certs::pki_dir}/nssdb",
  $nss_db_password_file = "${certs::pki_dir}/nss_db_password-file",
```

That also makes me wonder why we prefix eveything with `nss_db` if the class is already called `nssdb`. It ends up being `$certs::ssltools::nssdb::nss_db_*`

Perhaps we should make the class parameters unprefixed. We can keep compatibility variables in place until we update the code that uses them.

Reply:
You're right, this interface makes a lot more sense. I was sort of thinking about what is the minimum information I need to pass to make this work, and lost sight of, what are the core things I should be able to specify when creating an NSS DB.

Reply:
> That also makes me wonder why we prefix eveything with `nss_db` if the class is already called `nssdb`. It ends up being `$certs::ssltools::nssdb::nss_db_*`
> 
> Perhaps we should make the class parameters unprefixed. We can keep compatibility variables in place until we update the code that uses them.

I think this is a good point also. Probably best suited for a separate PR.